### ISSUE TITLE
unable to get past first bun dependancy

### ISSUE THREAD

**Issue by owocki** [carlbarrdahl/allo-sdk-research#20]
```
owocki@local /Users/owocki/Sites/allo-starter-kit/packages/kit~ % bun link @allo/kit --save                                              (git)-[main] :
bun link v1.1.9 (bb13798d)
  🔍 @allo/eslint-config [257/486]
error: package "@allo/eslint-config" not found registry.npmjs.org/@allo%2feslint-config 404
  🔍 @allo/typescript-config [260/488]
error: package "@allo/typescript-config" not found registry.npmjs.org/@allo%2ftypescript-config 404
warn: incorrect peer dependency "i18next@23.11.5"

warn: incorrect peer dependency "i18next@23.11.5"

warn: incorrect peer dependency "react@18.3.1"

error: package "@allo/kit" is not linked

To install a linked package:
   bun link my-pkg-name-from-package-json

Tip: the package name is from package.json, which can differ from the folder name.

```

**Comment by owocki** [comment]
let me try a `bun install` at project root first

**Comment by owocki** [comment]
i'm just a js noob :) need to learn the ropes

or just get a docker container up with all the dpenedncies pre-managed

**Comment by owocki** [comment]
> let me try a bun install at project root first

tried this no luck!

**Comment by hussmartinez** [comment]
TLDR:

For demo usage no need to use `bun link`:

1. Run `bun install` from the root folder, 

<img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/9665f187-faad-445a-990d-048d80979256/bb39de2b-848f-4716-bc48-937b640d6624?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS85NjY1ZjE4Ny1mYWFkLTQ0NWEtOTkwZC0wNDhkODA5NzkyNTYvYmIzOWRlMmItODQ4Zi00NzE2LWJjNDgtOTM3YjY0MGQ2NjI0IiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ._0-qzoDalstAgpS_nyejbdieLPXfyyCWbGvc6hS17O8 " alt="image.png" width="527" height="277" />

2. After that go to the demo folder and run `bun run dev`:

<img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/f2a217dc-1c7b-4498-b21a-c2af3943d089/10296c64-c2fa-4b0a-8d8a-8491833a711a?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS9mMmEyMTdkYy0xYzdiLTQ0OTgtYjIxYS1jMmFmMzk0M2QwODkvMTAyOTZjNjQtYzJmYS00YjBhLThkOGEtODQ5MTgzM2E3MTFhIiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.g4G7ChCRVTdvE24jBXYS-dEtm7cwXVZAcA-2w9EyZHw " alt="image.png" width="566" height="308" />

3. Now you should have the next.js app available in `localhost:3000`. The first time it will take a while to load the page, on my side was around 45 seconds…  time to take some deep breaths ;P

<img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/6a4174ca-e941-4930-8aaf-5a827ba1084d/eb12805f-362a-47c6-991d-7dd107367dbe?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS82YTQxNzRjYS1lOTQxLTQ5MzAtOGFhZi01YTgyN2JhMTA4NGQvZWIxMjgwNWYtMzYyYS00N2M2LTk5MWQtN2RkMTA3MzY3ZGJlIiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.1GbuRW58-HcANOFZ75WKyheIBqdrPX4bPV5XW0o6cPc " alt="image.png" width="1403" height="977" />

4. Now you can start checking the demo, or modify the code in demo page for testing the allo kit.

Explanation:

As far as I understood from the code:<br>Folder `/apps` contain examples of usage of `allo-starter-kit`, If you want to run them you don't need to use `bun link`, those apps already have the allo kit as a workspace dependency:

<img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/372ba6e6-8a76-445a-988c-5983f1751905/a2a3c1e0-f48c-4b3e-a5ec-9ae4faa38851?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS8zNzJiYTZlNi04YTc2LTQ0NWEtOTg4Yy01OTgzZjE3NTE5MDUvYTJhM2MxZTAtZjQ4Yy00YjNlLWE1ZWMtOWFlNGZhYTM4ODUxIiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.FyOEH6OGSwRHJrX28PIDwo5KqWciTK_6zSYTyVVLjZQ " alt="image.png" width="513" height="288" />

Just run `bun install` from the root folder and then run an app from it's own folder. (as in the TLDR above)<br><br>The command `bun link` should be used in a package that you wanna export (link) to another project as if it was installed from a package manage.

Once you run `bun link` from a package that has exports, like `packages/kit`:

<img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/03be47e8-2afe-499f-84c9-06a99a1de511/1e016140-9710-4d7c-891f-5b0cf910326b?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS8wM2JlNDdlOC0yYWZlLTQ5OWYtODRjOS0wNmE5OWExZGU1MTEvMWUwMTYxNDAtOTcxMC00ZDdjLTg5MWYtNWIwY2Y5MTAzMjZiIiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.mkFc-8Gl6TLWq7QnUkoqu_0_uUnACpWzj_ubfWeG25o " alt="image.png" width="513" height="540" />

You can go to another node/npm project folder and then if you run `bun link {package name} --save` you will get the same behavior as if you installed it. (notice that in this case the name of the package in `packages/kit/package.json` is `@allo/kit`.

This approach can be used with published or unpublished packages, as I used to use it to test locally when I was developing a design system component library without the need of keep publishing the package and then installing to debug, you can just link it, debug, apply the fix and then publish.<br><br>The README file is explaining how to do it for an external project.<br><br>The approach I took:

1. in `packages/link` I run `bun link`

   <img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/c676f77d-a34d-499c-bf6b-6541b28acc36/f2afe08e-9024-43be-b23d-5f895d3e27e5?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS9jNjc2Zjc3ZC1hMzRkLTQ5OWMtYmY2Yi02NTQxYjI4YWNjMzYvZjJhZmUwOGUtOTAyNC00M2JlLWIyM2QtNWY4OTVkM2UyN2U1IiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.jEDgf95m8KcntWJsUrJ7JNT0Ne4H8ELsjUWGc0fTecQ " alt="image.png" width="454" height="179" />
2. I created another folder called `test` outside of `allo-starter-kit`(it could be a React project, but just to troubleshoot your issue). <br>From the folder `test` I run `npm init` passing all the default options. This created an empty node project.

   <img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/12d957ba-0e1c-4f2f-8bbe-dd80455ad7cd/dbcfe507-8343-4695-b968-2ec3f0f2ce58?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS8xMmQ5NTdiYS0wZTFjLTRmMmYtOGJiZS1kZDgwNDU1YWQ3Y2QvZGJjZmU1MDctODM0My00Njk1LWI5NjgtMmVjM2YwZjJjZTU4IiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.V8lRl-R35Ypy3ZuHCijBZC8v_42F4VfOoGApwfe3cIo " alt="image.png" width="454" height="64" />
3. And from `test` folder I run `bun link @allo/kit --save`<br>And we get the package installed, noticed the `"@allo/kit": "link:@allo/kit"` line.

   <img src="https://uploads.linear.app/76ea5061-77c0-496e-9ffb-1445e581706a/225a0255-9a0c-46ad-b8c5-a53e09db4e21/6448f661-d3f5-48ad-b51b-6ac0b2ded029?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzc2ZWE1MDYxLTc3YzAtNDk2ZS05ZmZiLTE0NDVlNTgxNzA2YS8yMjVhMDI1NS05YTBjLTQ2YWQtYjhjNS1hNTNlMDlkYjRlMjEvNjQ0OGY2NjEtZDNmNS00OGFkLWI1MWItNmFjMGIyZGVkMDI5IiwiaWF0IjoxNzIxMzM1MTQ0LCJleHAiOjMzMjkxODk1MTQ0fQ.nT7G4lhPR1I6kSUO_sUIBrYczWD1pjr6ZD1r91mXAiI " alt="image.png" width="324" height="369" />

   4\. After that you can start developing your app using the `@allo/kit` as a linked dependency instead of as an installed dependency.

**Comment by owocki** [comment]
> Now you should have the next.js app available in localhost:3000. The first time it will take a while to load the page, on my side was around 45 seconds… time to take some deep breaths ;P

thanks for this!  i was able to get the app up but theres a exception here: 

<img width="1124" alt="Screenshot 2024-07-18 at 2 47 53 PM" src="https://github.com/user-attachments/assets/ca8e0d12-888d-4621-a19b-c654e837d464">

i dont seem to be able to run the `bun link` commands to solve it tho...


```
owocki@local /Users/owocki/Sites/allo-starter-kit/packages/kit~ % bun link @allo/kit --save
bun link v1.1.9 (bb13798d)
  🔍 @allo/typescript-config [1/2]
error: package "@allo/typescript-config" not found registry.npmjs.org/@allo%2ftypescript-config 404
  🔍 @allo/eslint-config [2/2]
error: package "@allo/eslint-config" not found registry.npmjs.org/@allo%2feslint-config 404
warn: incorrect peer dependency "react@18.3.1"

error: package "@allo/kit" is not linked

To install a linked package:
   bun link my-pkg-name-from-package-json

Tip: the package name is from package.json, which can differ from the folder name.


```

**Comment by owocki** [comment]
thanks @hussedev for the help.  updating readme to boy scout it for the next one => https://github.com/gitcoinco/allo-starter-kit/commit/a04b641ecd912ad1556e1688bab65e713f8c03ff

**Comment by hussmartinez** [comment]
No need of linking, the missing part in this case is to run `bun run build` from the `/kit` folder.<br><br>As an alternative to run the whole ecosystem in parallel (apps + kit in watch mode for dev porpoises) we could run `bun dev` from the root folder.

**Comment by owocki** [comment]
> As an alternative to run the whole ecosystem in parallel (apps + kit in watch mode for dev porpoises) we could run bun dev from the root folder.


maybe thats what should be in the readme?

**Comment by hussmartinez** [comment]
Maybe you are right, I don't know what the wishes were with this, but in my
case I would like to decide what runs and not, and normally I would never
run all ecosystem.
Only either:
- Storybook (for allo-kit development only) -> it would be nice to have a
script `bun storybook` from the root folder.
- Kit build watch mode (for allo-kit development combined with a project
that is using it with `bun link`) -> it would be nice to have a script `bun
build:watch` from the root folder.

Storybook it's very useful for development of component libraries like this
one and doesn't depend on the built code, it directly watches the codebase.

I think the only situation when we would want to run the whole ecosystem
(demo + opengraph + storybook + kit) it's in case that we are doing a demo
and want to have it all up and running quickly to demonstrate.

I am gonna create some new scripts in the package.json and update the
README, I will raise a PR later.

**Comment by owocki** [comment]
@carlbarrdahl here is the issue i had last week when u were out.  in case its useful for boy-scoutting the devex

**Comment by carlbarrdahl** [comment]
> in a package that you wanna export (link) to another project as if it was installed from a package manage.
> 
> Once you run `bun link` from a package

So there are two use-cases here:
1. git clone the allo starter kit with the intention to work on the allo kit codebase or its demo apps (this is for new features and fixes)
2. installing allo kit as a package in a separate app (this is what we want most devs to do) 


This video shows the second path (and uses link because the npm package hasn't been published yet):
https://www.loom.com/share/e7932c4da2e54902a3d9b66b69285f5e
